src/test/java/org/takes/tk/TkCORSTest.java

```diff
@@ -0,0 +1,127 @@
+import org.takes.rs.RsText;
+
+/**
+ * Test case for {@link TkCORS}.
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+public final class TkCORSTest {
+
+    /**
+     * TkCORS can handle a connection without origin in the request.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withoutOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+            .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(take);
+        final RsPrint response = new RsPrint(cors.act(new RqFake()));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * TkCORS can handle connections with correct domain on origin.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withCorrectDomainOnOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+            .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(
+            take,
+            "http://teamed.io",
+            "http://example.com"
+        );
+        final Set<String> headers = new HashSet<String>();
+        headers.add("Origin: http://teamed.io");
+        final Request req = new RqWithHeaders(new RqFake(), headers);
+        final RsPrint response = new RsPrint(cors.act(req));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * TkCors can't handle connections with wrong domain on origin.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withWrongDomainOnOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+        .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(
+            take,
+            "http://www.teamed.io",
+            "http://sample.com"
+        );
+        final Set<String> headers = new HashSet<String>();
+        headers.add("Origin: http://wrong.teamed.io");
+        final Request req = new RqWithHeaders(new RqFake(), headers);
+        final RsPrint response = new RsPrint(cors.act(req));
+        final String head = response.printHead();
+        MatcherAssert.assertThat(
+            "It was expected a 403 HTTP status.",
+            head,
+            Matchers.containsString("HTTP/1.1 403")
+        );
+        MatcherAssert.assertThat(
+            "Wrong value on Access Control Allow Credentias param.",
+            head,
+            Matchers.containsString("Access-Control-Allow-Credentials: false")
+        );
+    }
+
+    /**
+     * Checks if the given head contains text that identify it with HTTP 200
+     * status.
+     * @param head HTTP response head.
+     */
+    private void httpStatusOk(final String head) {
```

Comment:
@yegor256 I need your help on this method because the mentioned method avoid duplicated test assertion. How could I improve that?